nlp/desci_sense/shared_functions/prompting/jinja/keywords_extraction_template.py

```diff
@@ -2,18 +2,24 @@
+You are an expert annotator tasked with assigning keywords to social media posts. At the end of the prompt you will be given the text content of a post which you will extract and return the keywords from.
+
+The keywords should represent the most salient topics and concepts introduced in the post. The keywords should consist of two types: general topics and specific concepts. The topics keywords should indicate the general topics discussed by the posts and its references, like "#climate-change" or "#blockchain". The specific concept keywords should help people in the field better understand the post's contents (like "#monte-carlo-tree-search", or "#consensus-algorithms"). In addition to the above, you should also add a special '#academic' keyword if the post is related to academic research. Academic research is to be defined broadly, as anything that the author is likely to see as related to their research, including academic job offers, code repositories, blog posts and so on. If the post is clearly not related in any way to academic research, add '#not-academic'. {% if metadata_list|length > 0 %}  The post also includes references to external content. Details about the external references will be provided alongside the input post under "Reference Metadata". The keywords should also represent the external references! {% endif %}
 
 Rules:
 - You should choose up to {{ max_keywords}} keywords, plus the additional special academic/non-academic keyword!
-- Keywords should be prefixed with a hashtag, e.g., #AI
+- Keywords should be prefixed with a hashtag, e.g., #ai
+- Keywords should should be normalized to a consistent form of lowercase acronym e.g. 'AI' should be returned as 'ai' and separated words should be separated with a hyphen, e.g. 'ModelTheory' should be 'model-theory'    
+- do not separated compound words like blockchain. 
+
+- If there is not enough context in the post content or meta data, e.g. there are no words, return only the special keyword ( #academic or #not-academic)
 - Your final answer should be structured as follows:
     - Reasoning Steps: (your reasoning steps)
+    - Repeat the exact text of the post content given to you at the end of the prompt (should be the same text as the post content)
```

Comment:
I would flip the `Reasoning Steps` part with the `Repeat the exact text` part so that the repetition supports the reasoning and doesn't interfere with the strutured output format

Reply:
Wait, we said we wouldn't keep it in the prompt. It is not there currently, but it is an outdated version. We could add that and use it in the output to check for hallucinations try again x times, and fallback to no keywords.

Reply:
Hmm strange, not sure why I saw the outdated version then. I went over the new version and it looks good, there are some minor spelling issues I will push a fix for.
One question - what is https://github.com/Common-SenseMakers/sensemakers/blob/db105821f32353db184d74a1989210b23bc99d8d/nlp/desci_sense/shared_functions/prompting/jinja/keywords_extraction_template.py#L29 ? Can we remove that?